api/src/main/java/com/cloud/management/ManagementServerHost.java

```diff
@@ -14,19 +14,23 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
-package com.cloud.cluster;
+package com.cloud.management;
 
 public interface ManagementServerHost {
+    enum State {
+        Up, Starting, Down
```

Comment:
Do we use the Starting state for the management server?

Reply:
Not even down I think. At least I never saw it changing in the DB when I shutdown one (the shutdown process does process all the `stop()` method at all, but that's another problem).

Reply:
`Down`, I have seen in the database.

```
-------------------------------------------------------------------------------------------------------------
| name                  | state | version | service_ip     | service_port | last_update           | removed |
| --------------------- | ----- | ------- | -------------- | ------------ | --------------------- | ------- |
| ubuntu-acs            | Up    | 4.9.3.0 | 127.0.0.1      | 9090         | 2018-01-24 21:46:26.0 | <null>  |
| localhost.localdomain | Down  | 4.9.3.0 | 192.168.0.23 | 9090         | 2018-01-24 16:10:54.0 | <null>  |
-------------------------------------------------------------------------------------------------------------
```

If the state `Starting` is not used, we can remove it.

Reply:
Ok
`Starting` isn't use as far as idea can tell me, I'll remove it.

Reply:
aren't we planning for stopping to be implemented? i'd leave it for now, unless you want to be a purist ;)

Reply:
@marcaurele @rafaelweingartner 
As per @DaanHoogland  - I was also curious - we would need a state "Stopping" and another API command that can place the MS in "Stopping" state.

Reply:
@serverchief Yes, we will need much more work to implement a graceful shutdown of the server.